### ISSUE TITLE
Build from source fails (pkg-config fontconfig > bzip2)

### ISSUE THREAD

**Issue by curiousercreative** [zed-industries/zed#12778]
### Check for existing issues

- [X] Completed

### Repro
1. Follow [instructions for installing on Linux](https://github.com/zed-industries/zed/blob/main/docs/src/development/linux.md#building-zed-for-linux)
2. Run build
    ER: build success
    AR: build fails with error below

    >thread 'main' panicked at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/yeslogic-fontconfig-sys-3.2.0/build.rs:8:48:
    >  called `Result::unwrap()` on an `Err` value: "`PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=\"1\" PKG_CONFIG_ALLOW_SYSTEM_LIBS=\"1\" \"pkg-config\" \"--libs\" \"--cflags\" \"fontconfig\"` did not exit successfully: exit status: 1\nerror: could not find system library 'fontconfig' required by the 'yeslogic-fontconfig-sys' crate\n\n--- stdout\n\n\n--- stderr\nPackage bzip2 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `bzip2.pc'\nto the PKG_CONFIG_PATH environment variable\nPackage 'bzip2', required by 'freetype2', not found\n"

### Environment

```text
$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```

```
$ neofetch 
             /////////////
         /////////////////////
      ///////*767////////////////         curiouser@galp-curiouser 
    //////7676767676*//////////////       ------------------------ 
   /////76767//7676767//////////////      OS: Pop!_OS 22.04 LTS x86_64 
  /////767676///*76767///////////////     Host: Galago Pro galp5 
 ///////767676///76767.///7676*///////    Kernel: 6.8.0-76060800daily20240311- 
/////////767676//76767///767676////////   Uptime: 1 day, 4 hours, 30 mins 
//////////76767676767////76767/////////   Packages: 2981 (dpkg), 74 (flatpak) 
///////////76767676//////7676//////////   Shell: bash 5.1.16 
////////////,7676,///////767///////////   Resolution: 1920x1080 
/////////////*7676///////76////////////   DE: GNOME 42.9 
///////////////7676////////////////////   WM: Mutter 
 ///////////////7676///767////////////    WM Theme: Pop 
  //////////////////////'////////////     Theme: Pop-dark [GTK2/3] 
   //////.7676767676767676767,//////      Icons: Pop [GTK2/3] 
    /////767676767676767676767/////       Terminal: tilix 
      ///////////////////////////         CPU: 11th Gen Intel i7-1165G7 (8) @  
         /////////////////////            GPU: Intel TigerLake-LP GT2 [Iris Xe 
             /////////////                Memory: 21941MiB / 31945MiB 

                                                                  
                                                                  

```

**Comment by Moshyfawn** [comment]
Check the #11755 discussion to see if it could solve your problem.

**Comment by curiousercreative** [comment]
> Check the #11755 discussion to see if it could solve your problem.

The workaround you linked to worked for me.

**Comment by ReillyBrogan** [comment]
This is a packaging issue with your distribution. There is something that they will likely need to fix about their fontconfig.pc file which is either that it includes a `Requires` on `bzip2` either directly or transitively and yet a `bzip2.pc` pkgconfig file is either not part of the bzip2-devel (or whatever it's called) package or that fontconfig-devel does not depend on bzip2-devel either directly or transitively. There's nothing for Zed to do here except potentially add it as a troubleshooting step on the linux build docs.

**Comment by curiousercreative** [comment]
I actually don't even have the issue on a second machine running the same arch, distro and version, so it's an isolated issue with my install and I agree there's nothing to do here
